### PR TITLE
move inference examples to sub-pages, fix crosslinks

### PR DESCRIPTION
**Pull request by ahnitz**
This is a prerequisite for adding another example. This rearranges the examples to sub-pages and fixes the crosslinks when they reference each other. 

See https://www.atlas.aei.uni-hannover.de/work/ahnitz/projects/docs/infup1/latest/html/inference.html#examples

### CODE REVIEW

File: docs/inference/examples/analytic.rst [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+Several analytic distributions are available to run tests on. These can be run
+quickly on a laptop to check that a sampler is working properly.
+
+This example demonstrates how to sample a 2D normal distribution with the
+``emcee`` sampler. First, we create the following configuration file:
+
+.. literalinclude:: ../../../examples/inference/analytic-normal2d/normal2d.ini
+   :language: ini
+
+:download:`Download <../examples/inference/analytic-normal2d/normal2d.ini>`
```
Comment (cdcapano) [inline docs/inference/examples/analytic.rst]: Update the download link too. (Note that it's broken in your example.)

File: docs/inference/examples/analytic.rst [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+The number of dimensions of the distribution is set by the number of
+``variable_params``. The names of the parameters do not matter, just that just
+that the prior sections use the same names.
+
+Now run:
+
+.. literalinclude:: ../../../examples/inference/analytic-normal2d/run.sh
+   :language: bash
+
+:download:`Download <../examples/inference/analytic-normal2d/run.sh>`
```
Comment (cdcapano) [inline docs/inference/examples/analytic.rst]: Update download link.

File: docs/inference/examples/analytic.rst [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+``normal2d.hdf`` which contains the results. It should take about a minute to
+run. If you have a computer with more cores, you can increase the
+parallelization by changing the ``nprocesses`` argument.
+
+To plot the posterior distribution after the last iteration, run:
+
+.. literalinclude:: ../../../examples/inference/analytic-normal2d/plot.sh
+   :language: bash
+
+:download:`Download <../examples/inference/analytic-normal2d/plot.sh>`
```
Comment (cdcapano) [inline docs/inference/examples/analytic.rst]: Update download link.

File: docs/inference/examples/bbh.rst [hunk 4]
```diff
@@ -0,0 +1,207 @@
...
+   :language: ini
+
+:download:`Download <../../../examples/inference/priors/gw150914_like.ini>`
+
+In the ``[model]`` section we have set the model to be ``gaussian_noise``.  As
+described above, this is the standard model to use for CBC signals. It assumes
+that the noise is wide-sense stationary Gaussian noise. Notice that we had to
+provide additional arguments for the low frequency cutoff to use in each
+detector. These values are the lower cutoffs used for the likelihood integral.
+(See the |GaussianNoise| docs for details.)
```
Comment (cdcapano) [inline docs/inference/examples/bbh.rst]: The shortcuts for the `GaussianNoise` and `MarginalizedPhaseGaussianNoise` models are broken now. Copy the lines at the top of inference.rst and put them at the top of this file to fix that.

File: docs/inference/examples/bbh.rst [hunk 5]
```diff
@@ -0,0 +1,207 @@
...
+option) that is colored by the `Advanced LIGO updated design sensitivity curve
+<https://dcc.ligo.org/LIGO-T1800044/public>`_. (Note that this is ~3 times more
+sensitive than what the LIGO detectors were when GW150914 was detected.) The
+duration of data that will be analyzed is set by the
+``analysis-(start|end)-time`` arguments. These values are measured with respect
+to the ``trigger-time``. The analyzed data should be long enough such that it
+encompasses the longest waveform admitted by our prior, plus our timing
+uncertainty (which is determined by the prior on ``delta_tc``). Waveform
+duration is approximately determined by the total mass of a system. The lowest
+total mass (`= mass1 + mass2`) admitted by our prior is 20 solar masses. This
```
Comment (cdcapano) [inline docs/inference/examples/bbh.rst]: Add two back ticks around `= mass1 + mass2`.

File: docs/inference/examples/bbh.rst [hunk 6]
```diff
@@ -0,0 +1,207 @@
...
+   :language: ini
+
+:download:`Download <../../../examples/inference/samplers/emcee_pt-gw150914_like.ini>`
+
+Here, we will use 200 walkers and 20 temperatures. We will checkpoint (i.e.,
+dump results to file) every 2000 iterations. Since we have provided an
+``effective-nsamples`` argument and a ``[sampler-burn_in]`` section,
+``pycbc_inference`` will run until it has acquired 1000 independent samples
+after burn-in, which is determined by a combination of the :py:meth:`nacl
+<pycbc.inference.burn_in.MultiTemperedMCMCBurnInTests.nacl>` and
```
Comment (cdcapano) [inline docs/inference/examples/bbh.rst]: Change this to `<pycbc.inference.burn_in.MCMCBurnInTests.nacl>` to fix the cross link (sorry, this was a mistake I introduced when I wrote this).

File: docs/inference/examples/bbh.rst [hunk 7]
```diff
@@ -0,0 +1,207 @@
...
+:py:meth:`max_posterior
+<pycbc.inference.burn_in.MultiTemperedMCMCBurnInTests.max_posterior>` tests;
```
Comment (cdcapano) [inline docs/inference/examples/bbh.rst]: Change to `<pycbc.inference.burn_in.MCMCBurnInTests.max_posterior>` to fix this.

File: docs/inference/examples/bbh.rst [hunk 8]
```diff
@@ -0,0 +1,207 @@
...
+
+.. literalinclude:: ../../../examples/inference/bbh-injection/run.sh
+   :language: bash
+
+:download:`Download <../../../examples/inference/bbh-injection/run.sh>`
+
+
+Since we are generating waveforms and analyzing a 15 dimensional parameter
+space, this run will be much more computationally expensive than the analytic
+example above. We recommend running this on a cluster or a computer with a
```
Comment (cdcapano) [inline docs/inference/examples/bbh.rst]: Change "the analytic example above" to a link to the analytic example page.

**Comment by ahnitz** [comment]
@cdcapano check again.

File: docs/inference/examples/bbh.rst [hunk 9]
```diff
@@ -0,0 +1,210 @@
...
+:download:`Download <../../../examples/inference/samplers/emcee_pt-gw150914_like.ini>`
+
+Here, we will use 200 walkers and 20 temperatures. We will checkpoint (i.e.,
+dump results to file) every 2000 iterations. Since we have provided an
+``effective-nsamples`` argument and a ``[sampler-burn_in]`` section,
+``pycbc_inference`` will run until it has acquired 1000 independent samples
+after burn-in, which is determined by a combination of the :py:meth:`nacl
+<pycbc.inference.burn_in.MCMCBurnInTests.nacl>` and
+:py:meth:`max_posterior
+<<pycbc.inference.burn_in.MCMCBurnInTests.max_posterior>>` tests;
```
Comment (cdcapano) [inline docs/inference/examples/bbh.rst]: Shouldn't this just be a single `<` and `>`?

Comment (ahnitz) [reply docs/inference/examples/bbh.rst]: fixed

**Comment by ahnitz** [comment]
@cdcapano poke

**Pull request merged**